oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/incrementalstore/IncrementalFlatFileStoreStrategy.java

```diff
@@ -0,0 +1,181 @@
+import static org.apache.jackrabbit.oak.index.indexer.document.flatfile.FlatFileNodeStoreBuilder.OAK_INDEXER_MAX_SORT_MEMORY_IN_GB_DEFAULT;
+import static org.apache.jackrabbit.oak.index.indexer.document.flatfile.FlatFileStoreUtils.getMetadataFileName;
+import static org.apache.jackrabbit.oak.index.indexer.document.flatfile.FlatFileStoreUtils.getSortedStoreFileName;
+
+public class IncrementalFlatFileStoreStrategy implements IncrementalIndexStoreSortStrategy {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final String STORE_TYPE = "IncrementalFlatFileStore";
+    public static final String OAK_INDEXER_DELETE_ORIGINAL = "oak.indexer.deleteOriginal";
+    private final String beforeCheckpoint;
+    private final String afterCheckpoint;
+    private final PathElementComparator comparator;
+    private final NodeStateEntryWriter entryWriter;
+    private final File storeDir;
+    private final NodeStore nodeStore;
+    private final Compression algorithm;
+    private final Predicate<String> pathPredicate;
+    private final boolean deleteOriginal = Boolean.parseBoolean(System.getProperty(OAK_INDEXER_DELETE_ORIGINAL, "true"));
+    private final int maxMemory = Integer.getInteger(OAK_INDEXER_MAX_SORT_MEMORY_IN_GB, OAK_INDEXER_MAX_SORT_MEMORY_IN_GB_DEFAULT);
+    private long textSize = 0;
+    private long entryCount = 0;
+    private final Set<String> preferredPathElements;
+
+    public IncrementalFlatFileStoreStrategy(NodeStore nodeStore, @NotNull String beforeCheckpoint, @NotNull String afterCheckpoint, File storeDir,
+                                            Set<String> preferredPathElements, Compression algorithm, Predicate<String> pathPredicate, NodeStateEntryWriter entryWriter) {
+        this.nodeStore = nodeStore;
+        this.beforeCheckpoint = beforeCheckpoint;
+        this.afterCheckpoint = afterCheckpoint;
+        this.storeDir = storeDir;
+        this.algorithm = algorithm;
+        this.pathPredicate = pathPredicate;
+        this.entryWriter = entryWriter;
+        this.preferredPathElements = preferredPathElements;
+        this.comparator = new PathElementComparator(preferredPathElements);
+    }
+
+    @Override
+    public File createSortedStoreFile() throws IOException {
+        Stopwatch sw = Stopwatch.createStarted();
+        File file = new File(storeDir, getSortedStoreFileName(algorithm));
+        try (BufferedWriter w = FlatFileStoreUtils.createWriter(file, algorithm)) {
+            NodeState before = Objects.requireNonNull(nodeStore.retrieve(beforeCheckpoint));
+            NodeState after = Objects.requireNonNull(nodeStore.retrieve(afterCheckpoint));
+            EditorDiff.process(VisibleEditor.wrap(new IncrementalFlatFileStoreEditor(w, entryWriter, pathPredicate, this)), before, after);
+        }
+        String sizeStr = algorithm == null || algorithm.equals(Compression.NONE) ? "" : String.format("compressed/%s actual size", humanReadableByteCount(textSize));
```

Comment:
I don't think `algorithm` can be null here
```suggestion
        String sizeStr = algorithm.equals(Compression.NONE) ? "" : String.format("compressed/%s actual size", humanReadableByteCount(textSize));```

Reply:
Added @NotNull on constructor parameter and removed this condition.